### DISCUSSION TITLE
Need Help to compare Attributes

### DISCUSSION THREAD

**Discussion by naveenkulkarni029** [open-policy-agent/community#619] [Community]
Hello OPA World, I need Help in comparing attibutes between two objects. Below is my `JSON Data`

`{
  "output_groups": {
    "country_IN": true,
    "acc_group_wl": true,
    "organisationVerification": 1
  },
  "output_role_bindings_groups": "cm-users",
  "role": "CustomerManagement_ViewCustomers",
  "subject_attributes": {
    "country": "IN",
    "country_IN": true,
    "acc_group_wl": true,
    "email": "nbk111@gmail.com",
    "id": "145852",
    "organisationVerification": 1
  }
}`

I want to compare if all the attributes of the `output_groups `object are a part of the `subject_attributes`.

I tried with the `every `and `some `and then use `globmatch `[https://docs.styra.com/opa/rego-by-example/builtins/regex/globs_match](Link for the example) which did not match my requirement.

Reaching out to the community after multiple failed attempts.

Happy to hear back from you.

Thanks,
Naveen Kulkarni

**Comment by charlieegan3** [comment]
Hey, perhaps something like this: https://play.openpolicyagent.org/p/4rNangR9tf


```
package play

import rego.v1

default allow := false

allow if {
	every check_key, check_value in input.output_groups {
		input.subject_attributes[check_key] == check_value
	}
}
```

**Reply by naveenkulkarni029** [reply]
Hello Chalie, Thank you for your reply, meaning this every key, value here would iterate over the output_groups and check if the key and the value exists in the subject_attributes

**Reply by charlieegan3** [reply]
yes, and the value must also be the same as the value in the output_groups too.
